cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -279,10 +279,13 @@ class SequenceBuilder {
     RETURN_NOT_OK(offsets_.Finish(&offsets_array));
     const auto& offsets = checked_cast<const Int32Array&>(*offsets_array);
 
+    std::shared_ptr<Array> nones_array;
+    RETURN_NOT_OK(nones_.Finish(&nones_array));
+    const auto& nones = checked_cast<const NullArray&>(*nones_array);
```

Comment:
the way that `NullBuilder` is being used in this module doesn't look right; probably related to the NullBuilder changes above cc @pcmoritz. It looks like a BooleanBuilder should be used instead

Reply:
I reverted the `NullBuilder` changes and switched `nones_` to `BooleanBuilder`.

@pcmoritz could you check if 6753dcc is fine?